e2e-tests/verify-is-synced.sh

```diff
@@ -0,0 +1,139 @@
+  api_access = variant { enabled };
+  disable_api_if_not_fully_synced = variant { enabled }
+})"
+
+# Wait until the ingestion of stable blocks is complete.
+# The number of next block headers should be 3, the canister
+# should reject all requests.
+wait_until_stable_height 2 60
+
+# bitcoin_get_balance should panic.
+set +e
+MSG=$(dfx canister call bitcoin bitcoin_get_balance '(record {
+  network = variant { regtest };
+  address = "bcrt1qg4cvn305es3k8j69x06t9hf4v5yx4mxdaeazl8"
+})' 2>&1);
+set -e
+
+if ! [[ $MSG = *"Canister state is not fully synced."* ]]; then
+  echo "FAIL"
+  exit 1
+fi
+
+# bitcoin_get_utxos should panic.
+set +e
+MSG=$(dfx canister call bitcoin bitcoin_get_utxos '(record {
+  network = variant { regtest };
+  address = "bcrt1qxp8ercrmfxlu0s543najcj6fe6267j97tv7rgf";
+})' 2>&1);
+set -e
+
+if ! [[ $MSG = *"Canister state is not fully synced."* ]]; then
+  echo "FAIL"
+  exit 1
+fi
+
+# bitcoin_get_current_fee_percentiles should panic.
+set +e
+MSG=$(dfx canister call bitcoin bitcoin_get_current_fee_percentiles '(record {
+  network = variant { regtest };
+})' 2>&1);
+set -e
+
+if ! [[ $MSG = *"Canister state is not fully synced."* ]]; then
+  echo "FAIL"
+  exit 1
+fi
+
+dfx stop
+
+dfx start --background --clean
+
+# Deploy the canister that returns the blocks.
+dfx deploy --no-wallet e2e-verify-is-synced
+
+# Deploy the bitcoin canister, setting the blocks_source to be the source above.
+# And disabling 'disable_api_if_not_fully_synced'. Hence, it should not make 
+# influence behaviour of the canister. 
+dfx deploy --no-wallet bitcoin --argument "(record {
+  stability_threshold = 1;
+  network = variant { regtest };
+  blocks_source = principal \"$(dfx canister id e2e-verify-is-synced)\";
+  syncing = variant { enabled };
+  fees = record {
+    get_utxos_base = 0;
+    get_utxos_cycles_per_ten_instructions = 0;
+    get_utxos_maximum = 0;
+    get_balance = 0;
+    get_balance_maximum = 0;
+    get_current_fee_percentiles = 0;
+    get_current_fee_percentiles_maximum = 0;
+    send_transaction_base = 0;
+    send_transaction_per_byte = 0;
+  };
+  api_access = variant { enabled };
+  disable_api_if_not_fully_synced = variant { disabled }
+})"
+
+# Wait until the ingestion of stable blocks is complete.
+wait_until_main_chain_height 2 60
+
+BALANCE=$(dfx canister call bitcoin bitcoin_get_balance '(record {
+  network = variant { regtest };
+  address = "bcrt1qg4cvn305es3k8j69x06t9hf4v5yx4mxdaeazl8"
+})')
+
+if ! [[ $BALANCE = "(2 : nat64)" ]]; then
+  echo "FAIL"
+  exit 1
+fi
+
+# Verify that we are able to fetch the UTXOs of one address.
+UTXOS=$(dfx canister call bitcoin bitcoin_get_utxos '(record {
+  network = variant { regtest };
+  address = "bcrt1qg4cvn305es3k8j69x06t9hf4v5yx4mxdaeazl8"
+})')
+
+# The address has 2 UTXOs.
+if ! [[ $(num_utxos "$UTXOS") = 2 ]]; then
+  echo "FAIL"
+  exit 1
+fi
+
+dfx canister call bitcoin bitcoin_get_current_fee_percentiles '(record {
+  network = variant { regtest };
+})'
```

Comment:
I don't think this does anything actually. As in, this will call the endpoint but succeed no matter what. I think your intention here is to check that it's returning some values and not rejecting the request, which you'll need to do explicitly.

Reply:
It will panic if not synced hence the script will fail.

Reply:
I have added the check.